File: object/object/src/Lec02/policy/discount/PercentDiscountPolicy.java
package Lec02.policy.discount;

import Lec02.condition.discount.discountCondition;
import Lec02.Money;
import Lec02.Screening;

public class PercentDiscountPolicy extends DefaultDiscountPolicy {

    private double percent;

    public PercentDiscountPolicy(double percent, discountCondition... conditions) {
        super(conditions);
        this.percent = percent;
    }

    @Override
    protected Money getDiscountAmount(Screening screening) {
        return screening.getMovieFee().times(percent);
    }
}

File: object/object/src/java_to_kotlin/policy/discount/PercentDiscountPolicy.kt
package java_to_kotlin.policy.discount

import java_to_kotlin.Money
import java_to_kotlin.Screening
import java_to_kotlin.condition.discount.discountCondition

class PercentDiscountPolicy(private val percent: Double,
                            vararg conditions: discountCondition?) :
    DefaultDiscountPolicy(*conditions) {
    override fun getDiscountAmount(screening: Screening): Money? {
        return screening.movieFee?.times(percent)
    }
}
